src/Symfony/Component/Yaml/Dumper.php

```diff
@@ -98,11 +103,17 @@ public function dump($input, $inline = 0, $indent = 0, $flags = 0)
                     continue;
                 }
 
-                $willBeInlined = $inline - 1 <= 0 || !is_array($value) || empty($value);
+                $dumpObjectAsInlineMap = true;
+
+                if (Yaml::DUMP_OBJECT_AS_MAP & $flags && ($value instanceof \ArrayObject || $value instanceof \stdClass)) {
```

Comment:
Should this code be put in a private static method? is identical to what we have on line 86